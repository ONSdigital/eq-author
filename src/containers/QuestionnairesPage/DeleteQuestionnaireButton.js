import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Button from "components/Button";

import DeleteSvg from "./deleteQuestionnaireIcon.svg";

const DeleteQuestionnaireButton = ({ onClick }) => {
  const StyledButton = styled(Button)`
    width: 16px;
    height: 20px;
    padding: .5em;
    background: url(${DeleteSvg}) no-repeat;
    background-size: contain;
    border: 0;
    opacity: .7;
    
    &:hover { 
      opacity: 1;
      transition: opacity 150ms ease-in;
    }
  `;

  return <StyledButton title="Delete questionnaire" onClick={onClick} />;
};

DeleteQuestionnaireButton.propTypes = {
  onClick: PropTypes.func.isRequired
};

export default DeleteQuestionnaireButton;
