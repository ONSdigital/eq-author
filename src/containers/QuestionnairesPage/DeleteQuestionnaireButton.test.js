import React from "react";
import { shallow } from "enzyme";
import DeleteQuestionnaireButton from "./DeleteQuestionnaireButton";

describe("DeleteQuestionnaireButton", () => {
  let wrapper;
  let handleClick;

  beforeEach(() => {
    handleClick = jest.fn();
    wrapper = shallow(<DeleteQuestionnaireButton onClick={handleClick} />);
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should invoke onClick prop when delete button clicked", () => {
    wrapper.simulate("click");
    expect(handleClick).toHaveBeenCalled();
  });
});
