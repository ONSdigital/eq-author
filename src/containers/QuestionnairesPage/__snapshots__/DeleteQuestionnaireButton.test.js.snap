// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteQuestionnaireButton should render 1`] = `
.c0 {
  width: 16px;
  height: 20px;
  padding: 0.5em;
  background: url(deleteQuestionnaireIcon.svg) no-repeat;
  background-size: contain;
  border: 0;
  opacity: 0.7;
}

.c0:hover {
  opacity: 1;
  -webkit-transition: opacity 150ms ease-in;
  transition: opacity 150ms ease-in;
}

<Button
  className="c0"
  onClick={[Function]}
  title="Delete questionnaire"
/>
`;
